### ISSUE TITLE
TO,TP: server should not be allowed to be deleted when it has a DS assigned to it.

### ISSUE THREAD

**Issue by lbathina** [apache/trafficcontrol#4039]
## I'm submitting a ...

- [ ] bug report
- [ ] new feature / enhancement request
- [X] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [X] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

User is able to delete a server when it is assigned to a DS.
## Expected / new behavior:

User should not be able to delete a server when it is assigned to a DS. 

## Anything else:

This restriction should be in place to avoid having any DS unknowingly be not assigned with any caches. When a server is being deleted, all its associated links should be deleted before it could be. Which avoids any unknown implications.

**Comment by mitchell852** [comment]
this is not true. you can delete servers that are assigned to delivery services and always have been able to. why should you not be able to do this?

**Comment by mitchell852** [comment]
imagine if you could not do this. you have a server that happens to be assigned to 100 different delivery services. do you first have to unassign that server from all 100 delivery services before you can delete it? seems like a operations nightmare.

**Comment by ocket8888** [comment]
Sorta why I think it should maybe just be a confirmation dialog in TP - on top of the one we already have, or maybe extra cautionary text in the current one.

**Comment by mitchell852** [comment]
something like "the server you are deleting is currently assigned to 100 delivery services. Proceed?" I think that is fair.
